src/test/app/Clawback_test.cpp

```diff
@@ -0,0 +1,971 @@
+//------------------------------------------------------------------------------
+/*
+  This file is part of rippled: https://github.com/ripple/rippled
+  Copyright (c) 2023 Ripple Labs Inc.
+
+  Permission to use, copy, modify, and/or distribute this software for any
+  purpose  with  or without fee is hereby granted, provided that the above
+  copyright notice and this permission notice appear in all copies.
+
+  THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+  WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+  MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+  ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+  WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+  ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+  OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <ripple/basics/random.h>
+#include <ripple/json/to_string.h>
+#include <ripple/ledger/ApplyViewImpl.h>
+#include <ripple/protocol/Feature.h>
+#include <ripple/protocol/jss.h>
+#include <initializer_list>
+#include <test/jtx.h>
+#include <test/jtx/trust.h>
+
+namespace ripple {
+
+class Clawback_test : public beast::unit_test::suite
+{
+    template <class T>
+    static std::string
+    to_string(T const& t)
+    {
+        return boost::lexical_cast<std::string>(t);
+    }
+
+    // Helper function that returns the owner count of an account root.
+    static std::uint32_t
+    ownerCount(test::jtx::Env const& env, test::jtx::Account const& acct)
```

Comment:
I believe these two functions are common to a few other test suites like `NFToken`, `NFTokenBurn` and `Checks`. Given that the function definitions are identical, can we place these functions in a common utility file?

Reply:
I think that makes sense, but not sure exactly where that could be put. I think it makes sense to create a issue later to address/clean this separately.

Reply:
Thanks 😀 

I'll address this separately.